.github/workflows/_ci.yaml

```diff
@@ -399,6 +399,180 @@ jobs:
+        kind: Service
+        metadata:
+          name: jax-headless-svc
+        spec:
+          clusterIP: None # clusterIP must be None to create a headless service
+          selector:
+            job-name: ${{ github.run_id }}-${{ github.run_attempt }}-jax # must match Job name
+        ---
+        apiVersion: batch/v1
+        kind: Job
+        metadata:
+          name: ${{ github.run_id }}-${{ github.run_attempt }}-jax
+        spec:
+          completions: 2 # number of nodes
+          parallelism: 2 # number of nodes
+          completionMode: Indexed
+          template:
+            spec:
+              subdomain: jax-headless-svc # has to match Service name
+              restartPolicy: Never
+              containers:
+              - name: jax
+                image: ${{ needs.build-jax.outputs.DOCKER_TAG_FINAL }}
+                ports:
+                - containerPort: 3389
+                command:
+                - sh
+                - -c
+                - |
+                  install-efa.sh
+                  nsys-jax \
+                    --output=/opt/output/rank%q{JOB_COMPLETION_INDEX}.zip \
+                    -- \
+                    jax-nccl-test \
+                    --coordinator-address \
+                    ${{ github.run_id }}-${{ github.run_attempt }}-jax-0.jax-headless-svc:3389 \
+                    --distributed \
+                    --gpus-per-process=8 \
+                    --process-count=2 \
+                    --process-id=$JOB_COMPLETION_INDEX
+                  touch /opt/output/.done
+                env:
+                - name: XLA_FLAGS
+                  value: --xla_gpu_enable_command_buffer=
+                resources:
+                  limits:
+                    nvidia.com/gpu: 8
+                    vpc.amazonaws.com/efa: 32
+                volumeMounts:
+                - mountPath: /dev/shm
+                  name: shmem
+                - mountPath: /opt/output
+                  name: output
+              - name: upload
+                image: amazon/aws-cli
+                command:
+                - sh
+                - -c
+                - |
+                  while [[ ! -f /opt/output/.done ]]; do
+                    sleep 1
+                  done
+                  aws s3 cp \
+                    /opt/output/rank${JOB_COMPLETION_INDEX}.zip \
+                    s3://jax-toolbox-eks-output/${{ github.run_id }}-${{ github.run_attempt }}-rank${JOB_COMPLETION_INDEX}.zip
+                volumeMounts:
+                - mountPath: /opt/output
+                  name: output
+              imagePullSecrets:
+              - name: ${{ github.run_id }}-${{ github.run_attempt }}-token
+              volumes:
+              - name: output
+                emptyDir: {}
+              - name: shmem
+                emptyDir:
+                  medium: Memory
+                  sizeLimit: 8Gi
+    steps:
+    - name: Login to GitHub Container Registry
```

Comment:
Is this still needed, as the container is pulled by the K8s job instead of the CI job itself?

Reply:
I believe it's needed to generate the secret that is saved to Kubernetes in the next step.

Reply:
I see. Makes sense.